### DISCUSSION TITLE
Just moved from 4.0.2 to 4.2.0 - Two very small questions

### DISCUSSION THREAD

**Discussion by rebop** [Aris-t2/CustomCSSforFx#485] [Q&A]
HI there Aris-t2. Just spent about 4 hours moving to latest version as I wil lbe updating FF ESR soon. All wnet well, though tedious, except for tow small issues I cannot find.

1. Cannot find where to change the color of the background of the tab bar, and
2. In previous version, when I hovered the location field or search field a border would appear to highlight it. It does not in the new version and cannot find where that setting might be.

Would appreciate the pointers.

Thanks so much.

**Comment by rebop** [comment]
I found one more I cannot find, the spacing of items in menus, particulalry the bookmarks menu, is to much compared to the last version (4.0.2) I was using. I am sure that was a result of a tweak I cannot find. Images below of 4.0.2 vs 4.2.0:

![1bm old](https://user-images.githubusercontent.com/13785042/182027483-1106f103-5a78-4910-97ac-25efd610f6fa.jpg)

![2bm new](https://user-images.githubusercontent.com/13785042/182027485-8c4543d8-104a-4ea1-b532-6a7e70c4c5ec.jpg)

Thanks again.

**Comment by rebop** [comment]
@Aris-t2 should I have posted this in a different place? I see things have changed here a bit. Sorry if so. Would hope these are easy and quick tweaks. All worked in a previous version, I just cannot find out where I tweked them after searching every css file in use.

Thanks.

~Bob

**Reply by rebop** [reply]
Tried to find  way to move this to General Questions and could not.

**Comment by Aris-t2** [comment]
> Cannot find where to change the color of the background of the tab bar, and

There is no "option" for that in general, it only is available for "tabs below navigation bar" mode in `color_variables.css`.

This will work: 

```css
#TabsToolbar:not(:-moz-lwtheme){
  background-color: blue !important;
}
```

> In previous version, when I hovered the location field or search field a border would appear to highlight it. It does not in the new version and cannot find where that setting might be.

This was a default browser feature.

The code below will work: 

```css
#urlbar:not([focused="true"]):hover #urlbar-background,
#searchbar:not(:focus-within):hover {
  border: 1px solid Highlight !important;
}
```

> I found one more I cannot find, the spacing of items in menus, particularly the bookmarks menu, is to much compared to the last version (4.0.2) I was using. I am sure that was a result of a tweak I cannot find. Images below of 4.0.2 vs 4.2.0:

You have to enable `popup_compact_menus.css`. Inside that file you can change spacing be modifying the value of `--item_y_padding` variable.

**Reply by rebop** [reply]
Thanks.
Background color of the tab bar DEFINITELY adjusted from default in 4.0.2. Just cannot find it!

Will try all of this and back later today when I know the results. 

Thanks! As always....

**Comment by rebop** [comment]
OK first good news:

I tweaked the highlight and added to my_userChrome.css and this works like a charm:
#urlbar:not([focused="true"]):hover #urlbar-background,
#searchbar:not(:focus-within):hover {
  border: .5px solid grey !important;
}

BUT:
#TabsToolbar:not(:-moz-lwtheme){
  background-color: blue !important;
}
in my_UserChrome did nothing.

I believe the code in 4.0.2 that made the change for me was in custom_tab_color_settings and was:
--default_tabs_toolbar_background: #F0F0F0

That does not work in 4.2.0

And last, I had this in my_userChrome 4.0.2 which DID work but not now:

:root{
  --item_y_padding: 10px; /* old value: 2px */
  --y_padding_before_label: 28px !important; /* for popup_menus_space_before_label.css */
  --y_padding_before_label_bm: calc(var(--y_padding_before_label) - 2px ) !important; /* for popup_menus_space_before_label.css */
  --panel-padding: 0px !important;
  --arrowpanel-menuitem-margin: 0px !important;
  --arrowpanel-menuitem-padding: calc( var(--item_y_padding) + 1px) 2px !important;
  --panel-subview-body-padding: 0px !important;
}

No matter WHAT I try in popup_compact_menus.css which IS enabled has no effect.

Can we look at those two a little more?

Thanks again.

**Comment by 2BeeOrNot** [comment]
Try

#TabsToolbar:not(:-moz-lwtheme){
/* background, not background-color */
background: blue !important;
}

**Reply by rebop** [reply]
Thanks, but no change.

Do i need to mention that I have my tabs at the bottom of the window? Does that make a difference?

**Reply by Aris-t2** [reply]
It does. Use the variable to change the color: https://github.com/Aris-t2/CustomCSSforFx/blob/master/classic/config/color_variables.css#L19

**Comment by rebop** [comment]
Thanks, but tried everything I could find or think of to change in color_variables.css and no changes at all to background color. Tried changing the gradient to a color, etc. Perhaps you could be more specific?

And same with any Y padding which had no effect. And as posted, the lines I had in my_userChrome for 4.0.2 do not qwork in 4.2.0. Any idea what can be changed there to make the padding work?

Thanks....

**Reply by rebop** [reply]
OK, just tried Chrome from 4.2.0 in Firefox 91 where I am using 4.0.2. The tab bar background does not change, but the padding in menus does using the lines from my_UserChrome. So it works in older Firefox, but not 103. Hopefully that helps with the padding.

Same with the 	custom_tab_color_settings   	--default_tabs_toolbar_background: #F0F0F0;

Works in FF 91. Not in FF 103.

**Reply by rebop** [reply]
Ooops, last part of that not true. The custom_tab_color_settings modified as above and put in FF 91 does not work. Sorry.

**Reply by Aris-t2** [reply]
`userChrome.css` > enabled `@import "./css/tabs/tabs_below_navigation_toolbar.css`
`color_variables.css` > changed `--tabs_toolbar_color_tabs_not_on_top: linear-gradient(red,blue) !important;`

result
![image](https://user-images.githubusercontent.com/6013569/183687676-8482bded-f870-4712-9752-6750db8069d8.png)


`userChrome.css` > enabled `@import "./css/tabs/tabs_below_main_content.css`
`color_variables.css` > changed `--tabs_toolbar_color_tabs_not_on_top: linear-gradient(red,blue) !important;`

result
![image](https://user-images.githubusercontent.com/6013569/183687830-fa80565a-70ff-4de7-b931-0c66748ea159.png)

**Comment by rebop** [comment]
Thanks so much, but does not work for me. No idea why not:

![1](https://user-images.githubusercontent.com/13785042/183699779-29804a89-59b1-4831-ad9e-07b5501475f2.JPG)

![2](https://user-images.githubusercontent.com/13785042/183699822-db614b2c-5f5f-44d7-a5dd-c9053309733b.JPG)

Something must be overriding this?

To be honest. This is just a nice to have. Would love the change, but more important tightening up the menus.

Thanks again Aris-T2

(Sorry - replying to the email did not work as expected)

**Reply by Aris-t2** [reply]
Make a clean setup of [this projects files](https://github.com/Aris-t2/CustomCSSforFx/releases) only. Edit files like shown above.

**Reply by rebop** [reply]
Sorry to reply so late, but small medical foo got in the way.

Just did exactly this and no change in the color. Truly Has me totally baffled. Almost like something still in memory.

But as mentioned, this is just a "nice to have" Will miss it, but more iomportant to tighten up menus. Can we try to sleuth thayt one further?

Aside from these two things, everything is just exactly as I desire.

**Comment by rebop** [comment]
OK, just tried an experient.

Downloaded a fresh FF ESR 102.1 portable, setup VIRGIN. Added the test Chrome with ONLY your suggested changes.

Tab bar did not move to the bottom and color did not change.

Has to be some sort of clue. no?

**Reply by rebop** [reply]
My bad. Forgot to change toolkit.legacyUserProfileCustomizations.stylesheets to true  and after that, it does work in **a virgin FF install**. So now to try to figure out why it does not work in my profile.

**Reply by rebop** [reply]
Its a theme conflict. with default theme, it works. SO now I can eiother live with the change (likely) or try to find another theme to muy liking where it does work. Whew. At least understood now.

So if you would not mind pointing me where to poke to change menu spacings. I will try changes in the virgin install to see if my old changes worked there while I poke around some more.

**Reply by Aris-t2** [reply]
Most color changes are for default theme only. That is made sure by using `:not(:-moz-lwtheme)` rule.

Instead of this, you need the opposite rule:

```css
#TabsToolbar:-moz-lwtheme {
  background-color: blue !important;
}
```
or
```css
#TabsToolbar:-moz-lwtheme {
  background-image: linear-gradient(red,blue) !important;
}
```

**Comment by rebop** [comment]
OK, got the menu paddings. Conflict in my_UserChrome and the .css for popup. Works as expected now. Sorry these were all me. Took a bit to sort them all out.

~Bob

**Comment by rebop** [comment]
OK, all is good again. Until the next FF ESR major changes at least.
Would like to send a latte. Paypal link just did not look right from GitHub. Can I have your PayPal name Aris-T2?

Thanks for all the help.

**Reply by Aris-t2** [reply]
It is the correct Paypal name you see following the link. "Aris" is just an old nickname ;-)
You can also send to my mail `aris-addons [at] gmx  [dot] net`

Thanks.

**Comment by rebop** [comment]
Here's an odd one from FF 103 (which has MAJOR problems in my opinion).

Pretty ugly. Is there a fix to remove the highlighting box? No rush. Will be some time until I actually can use 103.

![1 save](https://user-images.githubusercontent.com/13785042/184439452-6b769b97-7fd9-49a7-a92e-ffb88a809b57.jpg)

**Reply by Aris-t2** [reply]
Sorry for the late reply. 

I don't know the correct id for downloads dialog atm, but this will solve the issue (tested on Fx 119 Nightly):

```css
dialog *:focus {
  -moz-user-focus: none !important;
}
```
